docs/onboarding_to_cluster.md

```diff
@@ -0,0 +1,151 @@
+# Onboarding to a cluster
+
+Currently we support 2 ways to request/perform onboarding to any supported clusters:
+
+1. Onboarding request - please use the **Onboarding to Cluster** issue template in [operate-first/support](https://github.com/operate-first/support) repository.
+2. Opening a PR with the desired changes. The rest of this doc focuses on this second option. We'd gladly welcome your contributions.
+
+## Prerequisites
+
+1. Please install our [toolbox](https://github.com/operate-first/toolbox) to have the developer setup ready automatically for you.
+2. Or you can install the tools manually. You'll need [kustomize](https://kustomize.io/), [sops](https://github.com/mozilla/sops) and [ksops](https://github.com/viaduct-ai/kustomize-sops).
+
+Please fork/clone the [operate-first/apps](https://github.com/operate-first/apps) repository. **During this whole setup, we'll be working within this repository.**
+
+## Recipe
+
+If you want to know more about the overall design please consult the ADR documentation at [operate-first/support](https://github.com/operate-first/blueprint).
+
+In general we store all the cluster-scoped resources in a `cluster-scope` kustomize application within this repository.
+
+## Adding namespaces
+
+### Base resources
+
+Please run:
+
+```sh
+bash hack/onboarding.sh NAMESPACE_NAME OWNER_TEAM OPTIONAL_PROJECT_DESCRIPTION
+```
+
+This script will create
+
+- A namespace in `cluster-scope/base/namespaces/$NAMESPACE_NAME`
+- A blank user group for the `$OWNER_TEAM` if it does not exist yet in the `cluster-scope/base/groups/$OWNER_TEAM`
+- An RBAC component for the project admin role `RoleBinding` in `cluster-scope/components/namespace-admin-rolebinding/$OWNER_TEAM` and maps it to the newly added namespace.
+
+### Enabling namespace deployment to a specific cluster
+
+```sh
+cd cluster-scope/overlays/TARGET_CLUSTER
+kustomize edit add resource ../../base/namespaces/$NAMESPACE
+```
+
+## Adding users
+
+Please select target cluster in `cluster-scope/overlays/TARGET_CLUSTER` and change directory to this folder. There you either see a `secret-generator.yaml` (encrypted users) file or `group-user_patch.yaml` file (plaintext)
+
+### Encrypted overlay patch for users
+
+We'll use sops/ksops for the encryption in here
+
+#### Creating new group encrypted overlay patch
+
+If this is the first time your team is adding a user group, please ensure the `.sops.yaml` file in the repository root contains your team denotation. If it doesn't please add a section like this:
+
+```yaml
+# .sops.yaml
+creation_rules:
+  - encrypted_regex: "^users$"
+    path_regex: "cluster-scope/.*YOUR_TEAM_PREFIX.*"
```

Comment:
I'm a little confused by the path_regex here, we haven't been using prefixes, should we be using them? As it stands, the group manifests are  are just named after a designated team name. E.g. for operate-first we say `operate-first.enc.yaml` and not `opf-group.enc.yaml`. We could just say "cluster-scope/.*YOUR_TEAM_NAME.*" and further explain that `YOUR_TEAM_NAME` must match the group manifest name.

Reply:
Yeah.. I wanted to have a single rule per team (which would be solved by the prefix), but I think it may not be the best idea to have these sops rules anymore at all... They are just too confusing and also their order matters (the first resolved rule gets used)...

I think this is relevant to the comment you made above as well, I've added the rule explanation, but I'm not satisfied with it very much.

What about we scratch the idea of adding rules to `.sops.yaml` all together and just encrypt the files with CLI?

```
sops -e --encrypted-regex="^users$" --pgp="0508677DD04952D06A943D5B4DC4116D360E3276, YOUR_TEAM_GPG_KEY" path/to/resource.yaml
```

The encrypted resource hold the encryption metadata anyways so any further edit on that file will keep the `encrypted_regex` the same as well as the used keys... WDYT?

Reply:
Yeah, this makes it easier to digest. Let's go with this.

Reply:
Discussed offline. Changing the docs to the solution described in the [last comment](https://github.com/operate-first/support/pull/40#discussion_r560835605)